src/modules/drawing/lib/drawingUtils.js

```diff
@@ -27,12 +29,36 @@ export function wrapWebmercatorCoords(coords, inPlace = false) {
             coords.forEach((coords) => wrapWebmercatorCoords(coords, true))
             return coords
         } else {
-            return wrapXCoordinate(coords, getProjection(WEBMERCATOR.epsg))
+            return wrapX(coords, getProjection(WEBMERCATOR.epsg))
         }
     } else {
         return Array.isArray(coords[0])
             ? coords.map((coords) => wrapWebmercatorCoords(coords, false))
-            : wrapXCoordinate(coords.slice(), getProjection(WEBMERCATOR.epsg))
+            : wrapX(coords.slice(), getProjection(WEBMERCATOR.epsg))
+    }
+}
+
+/**
+ * Wraps the provided coordinates in the world extents (i.e. the coordinate range that
+ * if equivalent to the wgs84 [-180, 180))
+ *
+ * @param {Array} coords The coordinates (or array of coordinates) to wrap
+ * @param {CoordinateSystem} projection Projection of the coordinates
+ * @param {boolean} inPlace If false, the original coordinates remain untouched and only a copy is modified
+ * @returns If "inPlace", then the same reference as "coords", else a reference to the modified copy
+ */
+export function wrapXCoordinates(coords, projection, inPlace = false) {
+    if (inPlace) {
+        if (Array.isArray(coords[0])) {
+            coords.forEach((coords) => wrapXCoordinates(coords, projection, true))
+            return coords
+        } else {
+            return wrapX(coords, getProjection(projection.epsg))
+        }
+    } else {
+        return Array.isArray(coords[0])
+            ? coords.map((coords) => wrapXCoordinates(coords, projection, false))
+            : wrapX(coords.slice(), getProjection(projection.epsg))
```

Comment:
this looks awfully the same as what is done above, with just a call to `slice()` before
Might be a good idea to structure this function so that we do things only once, such as (pseudo code)
```
let wrappedCoords = coords
if (!inPlace) {
    wrappedCoords = wrappedCoords.slice()
}
if (is array) {
    return wrappedCoords.map((coord) => wrapXCoordinates(coord, projection, inPlace))
}
return wrapX(wrappedCoords, getProjection(projection.epsg))
```

Reply:
This makes the code much easier to understand, thanks. I've also added unit test for this code for checking the inPlace and not inPlace.